src/compiler.ts

```diff
@@ -8700,27 +8724,114 @@ export class Compiler extends DiagnosticEmitter {
       }
       let fieldInstance = <Field>member;
       let fieldType = fieldInstance.type;
-      exprs[i + 1] = this.module.store( // TODO: handle setters as well
-        fieldType.byteSize,
-        this.module.local_get(tempLocal.index, this.options.nativeSizeType),
-        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT),
-        fieldType.toNativeType(),
-        fieldInstance.memoryOffset
+      storeObjectLiteralField(
+        fieldInstance,
+        fieldType,
+        this.compileExpression(values[i], fieldInstance.type, Constraints.CONV_IMPLICIT)
       );
+
+      // This member is no longer omitted, so delete from our omitted fields
+      omittedClassFieldMembers.delete(member.name);
     }
     this.currentType = classReference.type.nonNullableType;
     if (hasErrors) return module.unreachable();
 
+    // Iterate through the remaining omittedClassFieldMembers.
+    if (members) {
+      for(let omittedClassFieldMemberKey of omittedClassFieldMembers.keys()) {
+        let member = members.get(omittedClassFieldMemberKey);
+        
+        if(member) {
+
+          let fieldInstance = <Field>member;
+          let fieldType = fieldInstance.type;
+
+          if (fieldType.classReference) {
```

Comment:
```suggestion
          if (fieldType.is(TypeFlags.REFERENCE) && fieldType.classReference !== null) {
```
There may be types that have a `classReference` (at some point), but are not actually classes, like the wrapper class `I32` of `i32`.